module/apmstreadwayamqp/utils.go

```diff
@@ -0,0 +1,64 @@
+package apmstreadwayamqp
+
+import (
+	"fmt"
+	"go.elastic.co/apm/module/apmhttp/v2"
+	"go.elastic.co/apm/v2"
+	"strings"
+)
+
+var (
+	elasticTraceparentHeader = strings.ToLower(apmhttp.ElasticTraceparentHeader)
+	w3cTraceparentHeader     = strings.ToLower(apmhttp.W3CTraceparentHeader)
+	tracestateHeader         = strings.ToLower(apmhttp.TracestateHeader)
+	apmStreadwayAmqpPrefix   = "apmstreadwayamqp_transaction_"
+)
+
+func getMessageTraceparent(headers map[string]interface{}, header string) (apm.TraceContext, bool) {
+	headerValue, ok := headers[fmt.Sprintf("%s_%s", apmStreadwayAmqpPrefix, header)]
+	if !ok || headerValue == nil {
+		return apm.TraceContext{}, false
+	}
+	switch headerValue.(type) {
+	case string:
+		if trP, err := apmhttp.ParseTraceparentHeader(headerValue.(string)); err == nil {
+			return trP, true
+		}
+	default:
+		return apm.TraceContext{}, false
+	}
+	return apm.TraceContext{}, false
+}
+
+func getMessageTracestate(headers map[string]interface{}, header string) (apm.TraceState, bool) {
+	headerValue, ok := headers[fmt.Sprintf("%s_%s", apmStreadwayAmqpPrefix, header)]
```

Comment:
```suggestion
	headerValue, ok := headers[fmt.Sprintf("%s_%s", apmStreadwayAmqpPrefix, header)].(string)
```

Again, I think you can use a simpler type assertion here.